src/components/HelpText/index.js

```diff
@@ -0,0 +1,178 @@
+import React, { useRef, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useUniqueIdentifier, useDisclosure } from '../../libs/hooks';
+import InternalOverlay from '../InternalOverlay';
+import RenderIf from '../RenderIf';
+import { ESCAPE_KEY } from '../../libs/constants';
+import {
+    StyledTooltip,
+    StyledTitle,
+    StyledIconContainer,
+    StyledText,
+    StyledButton,
+} from './styled';
+import {
+    ErrorIcon,
+    InfoIcon,
+    QuestionIcon,
+    WarningIcon,
+    ErrorInverseIcon,
+    InfoInverseIcon,
+    QuestionInverseIcon,
+    WarningInverseIcon,
+} from './icons';
+import positionResolver from './helpers/positionResolver';
+
+const iconMap = {
+    question: <QuestionIcon />,
+    info: <InfoIcon />,
+    error: <ErrorIcon />,
+    warning: <WarningIcon />,
+};
+
+const inverseIconMap = {
+    question: <QuestionInverseIcon />,
+    info: <InfoInverseIcon />,
+    error: <ErrorInverseIcon />,
+    warning: <WarningInverseIcon />,
+};
+
+/**
+ * HelpText is a popup that displays information related to an element.
+ */
+const HelpText = props => {
+    const { id, title, text, variant, tabIndex, className, style } = props;
+
+    const triggerRef = useRef();
+    const helpTextId = useUniqueIdentifier('help-text');
+    const [isFocused, setIsFocused] = useState(false);
+    const isHoverTooltip = useRef(false);
+    const isClickTooltip = useRef(false);
+    const { isOpen, open: openOverlay, close: closeOverlay } = useDisclosure(false);
+
+    useEffect(() => {
+        if (isFocused) {
+            openOverlay();
+        } else {
+            closeOverlay();
+        }
+    }, [closeOverlay, isFocused, openOverlay]);
+
+    const handleBlur = () => {
+        if (!isClickTooltip.current) {
+            setIsFocused(false);
+        }
+    };
+
+    const handleButtonMouseLeave = () => {
+        if (!isFocused) {
+            setTimeout(() => {
+                if (!isHoverTooltip.current) closeOverlay();
+            });
+        }
+    };
+
+    const handleTooltipMouseDown = () => {
+        isClickTooltip.current = true;
+    };
+
+    const handleTooltipMouseUp = () => {
+        isClickTooltip.current = false;
+        triggerRef.current.focus();
+    };
+
+    const handleTooltipMouseEnter = () => {
+        isHoverTooltip.current = true;
+    };
+
+    const handleTooltipMouseLeave = () => {
+        isHoverTooltip.current = false;
+        if (!isFocused) {
+            closeOverlay();
+        }
+    };
+
+    const handleKeyPressed = event => {
+        if (event.keyCode === ESCAPE_KEY) {
+            event.preventDefault();
+            closeOverlay();
+        }
+    };
+
+    const icon = iconMap[variant] || iconMap.info;
+    const inverseIcon = inverseIconMap[variant] || inverseIconMap.info;
+
+    return (
+        <div id={id} className={className} style={style}>
```

Comment:
I think we can't need to use a `div` wrapper here, instead we can use a Fragment `<>` and add the `className` `style` and `id` to the `StyledButton` element